checks/linux.py

```diff
@@ -0,0 +1,176 @@
+                We generally recommend using \"Window Capture (Xcomposite)\" on X11, as \"Display Capture (XSHM)\" can introduce bottlenecks depending on your setup.<br><br>
+                From a technical standpoint a Desktop Environment should not advertise the availability of these captures if they are not implemented. This behaviour can therefore be considered a bug in your Desktop Environment."""]
+
+    return [LEVEL_INFO, "X11",
+            "Window Capture is available via Xcomposite, while Display Capture is available via XSHM. We generally recommend sticking to \"Window Capture (Xcomposite)\" since \"Display Capture (XSHM)\" can introduce bottlenecks depending on your setup."]
+
+
+def checkDesktopEnvironment(lines):
+    isDistroNix = search('Distribution:', lines)
+    isFlatpak = search('Flatpak Runtime:', lines)
+
+    if (len(isDistroNix) <= 0) and (len(isFlatpak) <= 0):
+        return
+
+    desktopEnvironmentLine = search('Desktop Environment:', lines)
+    desktopEnvironment = desktopEnvironmentLine[0].split()
+    desktopEnvironment = desktopEnvironment[3:]
+    desktopEnvironment = ' '.join(desktopEnvironment)
+
+    if (len(desktopEnvironment) > 0):
+        return [LEVEL_INFO, desktopEnvironment, '']
+
+
+def checkMissingModules(lines):
+    isDistroNix = search('Distribution:', lines)
+
+    if (len(isDistroNix) <= 0):
+        return
+
+    modulesMissingList = []
+    modulesCheckList = ('obs-browser.so', 'obs-websocket.so', 'vlc-video.so')
+
+    for module in modulesCheckList:
+        if not search(module, lines):
+            modulesMissingList.append(module)
+
+    if len(modulesMissingList):
+        modulesMissingString = str(modulesMissingList)
+        modulesMissingString = modulesMissingString.replace("', '", "</li><li>")
+        modulesMissingString = modulesMissingString[2:]
+        modulesMissingString = modulesMissingString[:-2]
+
+        return [LEVEL_INFO, "Missing Modules (" + str(len(modulesMissingList)) + ")",
+                """You are missing the following default modules:<br><ul><li>""" + modulesMissingString + "</li></ul>"]
+
+
+def checkLinuxVCam(lines):
+    isDistroNix = search('Distribution:', lines)
+
+    if (len(isDistroNix) <= 0):
+        return
+
+    hasV4L2Module = search('v4l2loopback not installed', lines)
+
+    if len(hasV4L2Module) > 0:
+        return [LEVEL_INFO, "VCam not available",
+                """Using the Virtual Camera requires the <code>v4l2loopback</code> kernel module to be installed.<br>
+                If required, please refer to our <a href="https://github.com/obsproject/obs-studio/wiki/install-instructions#prerequisites-for-all-versions">Install Instructions</a> on how to install this on your distribution.<br>
```

Comment:
Perhaps we should link to https://obsproject.com/kb/virtual-camera-troubleshooting instead (which also links to https://github.com/obsproject/obs-studio/wiki/install-instructions#prerequisites-for-all-versions).

Reply:
I don't have a strong opinion. This is better than no message.